examples/cloudml-sentiment-analysis/scoring.py

```diff
@@ -0,0 +1,237 @@
+  service = discovery.build('ml', 'v1')
+  name = 'projects/{}/models/{}'.format(project, model)
+
+  if version is not None:
+    name += '/versions/{}'.format(version)
+
+  response = service.projects().predict(
+      name=name,
+      body={'instances': instances}
+  ).execute()
+
+  if 'error' in response:
+    raise RuntimeError(response['error'])
+
+  return response['predictions']
+
+
+def analyze(probas, target):
+  """Analyzes predictions and returns results.
+
+  Computes different metrics (specified by `constants.METRICS`) comparing
+  predictions to true labels.
+
+  Args:
+    probas: `np.array` with predicted probabilities.
+    target: `np.array` of `int` with true labels.
+
+  Returns:
+    Dictionary of `str` to `float` mapping metric names to the corresponding
+      scores.
+  """
+
+  results = {}
+  for metric_type, sub_metrics in constants.METRICS.iteritems():
+    for metric_name in sub_metrics:
+      metric = getattr(metrics, metric_name)
+
+      results[metric_name] = metric(
+          target,
+          (probas if metric_type == constants.CONTINUOUS_TYPE
+           else probas > constants.ACCURACY_THRESHOLD))
+  return results
+
+
+def run(project, model, size, input_path, batch_size, random_seed=None):
+  """Runs prediction job on sample of labelled reviews and analyzes results.
+
+  Args:
+    project: `str`, GCP project id.
+    model: `str`, name of Cloud ML Engine model.
+    size: `int`, number of reviews to process.
+    input_path: `str`, path to input data (reviews).
+    batch_size: `int`, size of predictions batches.
+    random_seed: `int`, random seed for sub-sample selection.
+
+  Returns:
+    Dictionary of `str` to `float` mapping metric names to the corresponding
+      scores.
+
+  Raises:
+    ValueError: If the total number of review found don't match the number of
+      files.
+    ValueError: If the size of output is not greater than `0`.
+    ValueError: If the number of predictions returned by API dont match input.
+  """
+
+  if random_seed is not None:
+    np.random.seed(random_seed)
+
+  def _get_probas(subdir):
+    """Computes predicted probabilities from records in input directory."""
+
+    instances = format_input(os.path.join(input_path, subdir), size)
+    # Checks that the number of records matches the number of files (expected
+    # exactly one review per file.
+    if len(instances) != size:
+      raise ValueError(
+          'Number of reviews found dont match the number of files.')
+
+    probas = collections.defaultdict(lambda: [])
+    step = int(size / batch_size) if batch_size else size
+    start = 0
+    failed_predictions = 0
+    while start < len(instances):
+      to_predict = instances[start:(start+step)]
+      try:
+        predictions = predict_json(project, model, to_predict)
+      except KeyboardInterrupt:
+        raise
+      except:  # pylint: disable=bare-except
+        logging.info('Error.')
```

Comment:
I think it would be a good idea to actually log the exception, not just "Error".